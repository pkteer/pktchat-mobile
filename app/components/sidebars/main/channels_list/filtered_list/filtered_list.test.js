// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import Preferences from '@mm-redux/constants/preferences';
import {shallowWithIntl} from '@test/intl-test-helper';

import FilteredList from './filtered_list';

describe('ChannelsList FilteredList', () => {
    const baseProps = {
        actions: {
            getProfilesInTeam: jest.fn(),
            makeGroupMessageVisibleIfNecessary: jest.fn(),
            searchChannels: jest.fn(),
            searchProfiles: jest.fn(),
        },
        onSelectChannel: jest.fn(),
        testID: 'main.sidebar.channels_list.filtered_list',
        channels: {},
        currentTeam: {},
        currentUserId: 'current-user-id',
        currentChannel: {},
        groupChannelMemberDetails: {},
        teammateNameDisplay: 'teammate-name-display',
        otherChannels: [],
        archivedChannels: [],
        profiles: {},
        teamProfiles: {},
        searchOrder: [],
        pastDirectMessages: [],
        restrictDms: false,
        statuses: {},
        styles: {},
        term: 'term',
        theme: Preferences.THEMES.denim,
        isLandscape: false,
    };

    test('should match snapshot', () => {
        const wrapper = shallowWithIntl(<FilteredList {...baseProps}/>);

        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
