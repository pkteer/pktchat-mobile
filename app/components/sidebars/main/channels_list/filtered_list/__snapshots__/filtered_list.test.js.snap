// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelsList FilteredList should match snapshot 1`] = `
<FilteredList
  actions={
    Object {
      "getProfilesInTeam": [MockFunction],
      "makeGroupMessageVisibleIfNecessary": [MockFunction],
      "searchChannels": [MockFunction],
      "searchProfiles": [MockFunction],
    }
  }
  archivedChannels={Array []}
  channels={Object {}}
  currentChannel={Object {}}
  currentTeam={Object {}}
  currentUserId="current-user-id"
  groupChannelMemberDetails={Object {}}
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  isLandscape={false}
  onSelectChannel={[MockFunction]}
  otherChannels={Array []}
  pastDirectMessages={Array []}
  profiles={Object {}}
  restrictDms={false}
  searchOrder={Array []}
  statuses={Object {}}
  styles={Object {}}
  teamProfiles={Object {}}
  teammateNameDisplay="teammate-name-display"
  term="term"
  testID="main.sidebar.channels_list.filtered_list"
  theme={
    Object {
      "awayIndicator": "#ffbc1f",
      "buttonBg": "#1c58d9",
      "buttonColor": "#ffffff",
      "centerChannelBg": "#ffffff",
      "centerChannelColor": "#3f4350",
      "codeTheme": "github",
      "dndIndicator": "#d24b4e",
      "errorTextColor": "#d24b4e",
      "linkColor": "#386fe5",
      "mentionBg": "#ffffff",
      "mentionColor": "#1e325c",
      "mentionHighlightBg": "#ffd470",
      "mentionHighlightLink": "#1b1d22",
      "newMessageSeparator": "#cc8f00",
      "onlineIndicator": "#3db887",
      "sidebarBg": "#1e325c",
      "sidebarHeaderBg": "#192a4d",
      "sidebarHeaderTextColor": "#ffffff",
      "sidebarTeamBarBg": "#14213e",
      "sidebarText": "#ffffff",
      "sidebarTextActiveBorder": "#5d89ea",
      "sidebarTextActiveColor": "#ffffff",
      "sidebarTextHoverBg": "#28427b",
      "sidebarUnreadText": "#ffffff",
      "type": "Denim",
    }
  }
/>
`;
